Stp.TestingApi/Controllers/TaskController.cs

```diff
@@ -25,44 +26,53 @@ public TaskController(TestingDbContext db)
 
         /* https://localhost:5001/api/Task/GetTasksByCategory */
         [HttpGet(nameof(GetTasksByCategory))]
-        public IEnumerable<Contracts.TaskDto> GetTasksByCategory(long taskCategoryId)
+        public IEnumerable<TaskDto> GetTasksByCategory(long taskCategoryId)
         {
+
             var res = _db.TaskList
                 .Where(x => x.CategoryId == taskCategoryId)
                 .Include(x => x.MultichoiceAnswers)
-                .Select(x => new Contracts.TaskDto()
+                .Include(x => x.TaskAndSkills)
+                .ThenInclude(x => x.Skill)
+                .Select(x => new TaskDto()
                 {
-                    TaskSummary = new Contracts.TaskSummaryDto()
+                    TaskSummary = new TaskSummaryDto()
                     {
                         Id = x.Id,
                         Name = x.Name,
                         Type = x.Type,
                         Points = x.Points,
-                        Position = x.Position,
                         DurationMinutes = x.DurationMinutes,
                         Complexity = x.Complexity,
-                        Skills = new List<string>()
+                        Position = x.Position
                     },
 
-                    MultichoiceTaskInfo = new Contracts.MultichoiceTaskInfoDto()
+                    MultichoiceTaskInfo = new MultichoiceTaskInfoDto()
                     {
-                        Answers = x.MultichoiceAnswers.Select(a => new Contracts.MultichoiceTaskAnswerDto()
+                        Question = String.Empty,
+                        Answers = x.MultichoiceAnswers.Select(a => new MultichoiceTaskAnswerDto()
                         {
                             Id = a.Id,
                             Name = a.Name,
                             IsCorrect = a.IsCorrect
                         }).ToList()
                     },
 
-                    CodingTaskInfo = new Contracts.CodingTaskInfoDto()
-                }); ;       
+                    Skills = x.TaskAndSkills.Select(ts => new SkillDto() 
+                    { 
+                        Id = ts.Skill.Id, 
+                        Name = ts.Skill.Name
+                    }).ToList(),
+
+                    CodingTaskInfo = new CodingTaskInfoDto()
+                });     
 
             return res;
         }
 
         /* https://localhost:5001/api/Task/AddTask */
         [HttpPost(nameof(AddTask))]
-        public Contracts.TaskDto AddTask(long taskCategoryId, [FromBody] Contracts.TaskDto task)
+        public ActionResult<TaskDto> AddTask(long taskCategoryId, [FromBody] TaskDto task)
```

Comment:
In fact it not neccesarry to add ActionResult when returning an object. You can return just object itself. It will mean the same.

Reply:
Also it would be nice to add [ProducesResponseType] and [Consumes] attributes.

Reply:
> In fact it not neccesarry to add ActionResult when returning an object. You can return just object itself. It will mean the same.

It allows to process errors correctly. With bad request not with null.